nixos/modules/services/misc/metabase.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.metabase;
+
+  inherit (lib) mkEnableOption mkIf mkOption;
+  inherit (lib) optional optionalAttrs types;
+
+  dataDir = "/var/lib/metabase";
+
+in {
+
+  options = {
+
+    services.metabase = {
+      enable = mkEnableOption "Metabase service";
+
+      listen = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          description = ''
+            IP address that Metabase should listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 3000;
+          description = ''
+            Listen port for Metabase.
+          '';
+        };
+      };
+
+      ssl = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to enable SSL (https) support.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8443;
+          description = ''
+            Listen port over SSL (https) for Metabase.
+          '';
+        };
+
+        keystore = mkOption {
+          type = types.nullOr types.path;
+          default = "${dataDir}/metabase.jks";
+          example = "/etc/secrets/keystore.jks";
+          description = ''
+            <link xlink:href="https://www.digitalocean.com/community/tutorials/java-keytool-essentials-working-with-java-keystores"/>Java KeyStore</link> file containing the certificates.
```

Comment:
> ~/n/n/d/manual> nix-shell -p gnumake --run make
nix-build ../../release.nix \
        --attr manualGeneratedSources.x86_64-linux \
        --out-link ./generated
these derivations will be built:
  /nix/store/lxy26hkslkvw4smdmala9xy8nhhip2a6-options-db.xml.drv
  /nix/store/c1bfqi2ki7fw6xgihdb8vzvz0wbkam3z-generated-docbook.drv
building '/nix/store/lxy26hkslkvw4smdmala9xy8nhhip2a6-options-db.xml
.drv'...
intermediate.xml:29764: parser error : Opening and ending tag mismat
ch: para line 29764 and link
rials/java-keytool-essentials-working-with-java-keystores"/>Java Key
Store</link>

Reply:
Good catch.